src/writer/blocks/writeraddtokg.py

```
@@ -94,20 +94,30 @@ def run(self):
+                lambda: writer.ai.retrieve_graph(graph_id)
+            )
             
             # Add files to the graph
             for prepared_file in prepared_files:
-                file = writer.ai.upload_file(prepared_file.get("data"),
-                                             prepared_file.get("type"),
-                                             prepared_file.get("name"))
-                graph.add_file(file)
+                file = self._retry_on_auth_error(
+                    lambda: writer.ai.upload_file(
+                        prepared_file.get("data"),
```

Ruff (0.14.7)

Function definition does not bind loop variable `prepared_file`

(B023)